datastore/datastore.go

```diff
@@ -0,0 +1,322 @@
+	}
+
+	return kinds, nil
+}
+
+//DatastoreData converts a []DatastoreKind to a []map[string]interface{} to be used for the final output to new relic
+func DatastoreData(kinds []DatastoreKind, projectID string) []map[string]interface{} {
+	var datastoreData []map[string]interface{}
+	for _, k := range kinds {
+		datastoreData = append(datastoreData, map[string]interface{}{
+			"event_type":                         "DatastoreSample",
+			"provider":                           "datastoreQuery",
+			"datastoreQuery.builtinIndexBytes":   k.BuiltinIndexBytes,
+			"datastoreQuery.builtinIndexCount":   k.BuiltinIndexCount,
+			"datastoreQuery.compositeIndexBytes": k.CompositeIndexBytes,
+			"datastoreQuery.compositeIndexCount": k.CompositeIndexCount,
+			"datastoreQuery.entityBytes":         k.EntityBytes,
+			"datastoreQuery.bytes":               k.Bytes,
+			"datastoreQuery.count":               k.Count,
+			"datastoreQuery.kindName":            k.KindName,
+			"datastoreQuery.projectId":           projectID,
+			"datastoreQuery.timestamp":           k.Timestamp.Unix(),
+		})
+	}
+	return datastoreData
+}
+
+func fatalIfErr(log *logrus.Logger, err error) {
+	if err != nil {
+		log.WithError(err).Fatal("can't continue")
+	}
+}
+
+// ConnectDatastore establishes a datastore.Client from a base64 encoding JSON credentials file.
+func ConnectDatastore(base64Config string) (*datastore.Client, string, error) {
+	jsonConfig, err := base64.StdEncoding.DecodeString(base64Config)
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to decode datastore credentials: %v", err)
+	}
+	projectId, err := jsonparser.GetString(jsonConfig, "project_id")
+	if err != nil {
+		return nil, "", fmt.Errorf("failed to determine project_id from credentials file: %v", err)
+	}
+
+	ctx := context.Background()
+	creds, err := google.CredentialsFromJSON(ctx, jsonConfig, datastore.ScopeDatastore)
+	if err != nil {
+		return nil, "", err
+	}
+	c, err := datastore.NewClient(ctx, projectId, option.WithCredentials(creds))
+	if err != nil {
+		return nil, "", err
+	}
+	return c, projectId, err
+}
+
+//StackdriverResp gets the data of the wanted metric from the stackdriver API. Start time is set at -3 minutes to act as a Timestamp
+//as data from stackdriver is always 3 minutes old and refreshed every 1 minute. This timing also ensures we only ever get 1 point
+//back at a time.
+func StackdriverResp(projectId string, metric string) (*monitoring.ListTimeSeriesResponse, error) {
+	ctx := context.Background()
+	jsonConfig, err := base64.StdEncoding.DecodeString(base64Config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode datastore credentials: %v", err)
+	}
+
+	projectId, err = jsonparser.GetString(jsonConfig, "project_id")
+	if err != nil {
+		return nil, fmt.Errorf("failed to determine project_id from credentials file: %v", err)
+	}
+
+	creds, err := google.CredentialsFromJSON(ctx, jsonConfig, monitoring.MonitoringScope)
+
+	hc := oauth2.NewClient(ctx, creds.TokenSource)
```

Comment:
Do we have to create a new oauth client everytime? Thinking it might be worthwile to setup a Stackdriver client similar to how you have the datastore one